tests.stdenv.hooks: add more tests

Artturin opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Artturin committed:
oniguruma,jq: add artturin as maintainer

Artturin committed:
jq,oniguruma: make suitable for inclusion in common-path.nix

Artturin committed:
tests.stdenv: add test-golden-example-structuredAttrs

examples copied from https://nixos.mayflower.consulting/blog/2020/01/20/structured-attrs/

nix has tests for structuredAttrs but i figured it'd be good to have a
golden test here too

https://github.com/NixOS/nix/blob/master/tests/structured-attrs.nix
https://github.com/NixOS/nix/blob/master/tests/structured-attrs-shell.nix

Artturin committed:
tests: move stdenv hook tests to stdenv.hooks

Artturin committed:
move-docs.sh: update comment

_multioutDocs is in multiple-outputs.sh

Artturin committed:
tests.stdenv.hooks: add more tests

roberth commented on pkgs/development/libraries/oniguruma/default.nix:
```diff
@@ -1,24 +1,27 @@
-{ lib, stdenv, fetchFromGitHub, autoreconfHook }:
+{ lib, stdenv, fetchurl, autoreconfHook }:
+
+# Note: do not use fetchpatch or fetchFromGitHub to keep this package available in __bootPackages
```
Maybe put this near `src`, and add `passthru.tests` so ofborg will catch a problem if it arises?

SuperSandro2000 replied:
I would rather lift the tests one lever higher than limiting fetchpatch and so on in more packages.

roberth replied:
The developer experience of stdenv is worth more than lifting fetcher function restrictions in only two packages.

roberth commented on pkgs/development/tools/jq/default.nix:
```diff
@@ -1,29 +1,24 @@
 , oniguruma
 }:
 
+# Note: do not use fetchpatch or fetchFromGitHub to keep this package available in __bootPackages
```
Same

roberth commented on pkgs/test/stdenv/hooks.nix:
```diff
@@ -0,0 +1,130 @@
+{ stdenv, pkgs }:
+
+# ordering should match defaultNativeBuildInputs
```
I'd just embrace the chaos and/or sort alphabetically.
This is about the attrs below?

Artturin replied:
i sorted defaultNativeBuildInputs alphabetically instead

roberth reviewed (commented):
I think testing is important enough to impose one or two restrictions on these packages.

Artturin committed:
build-support/setup-hooks: change shebang to shellcheck directive

Artturin committed:
stdenv/generic: fix todo

Artturin committed:
audit-tmpdir.sh: fix on darwin

compress-man-pages.sh uses `find ... | while ...; do...` so this works
on darwin

vcunat says the following in https://github.com/NixOS/nixpkgs/commit/94d164dd7#r22030369

> Unfortunately, process substitution doesn't work during early darwin bootstrap. http://hydra.nixos.org/build/52433342 We've run into that before.
> I think it should work to have `find ... | while ...; do...`

Artturin committed:
stdenv: remove now unneeded linux conditional

Artturin committed:
stdenv: sort defaultNativeBuildInputs alphabetically

roberth commented on pkgs/stdenv/generic/default.nix:
```diff
@@ -57,28 +57,21 @@ argsStdenv@{ name ? "stdenv", preHook ? "", initialPath
-    ];
+      ../../build-support/setup-hooks/set-source-date-epoch-to-latest.sh
+      ../../build-support/setup-hooks/strip.sh
+    ] ++ lib.optionals hasCC [ cc ];
```
This reordering may have side effects.
I think I meant to only suggest sorting the tests. Doing this makes the change a lot riskier.
Also I think you've accidentally removed the isLinux conditional.

Artturin replied:
> This reordering may have side effects.

checking for those rn


>  Also I think you've accidentally removed the isLinux conditional.

that's intentional , because i fixed the fixme and want to not have a unneeded conditional

Artturin replied:
built `sway` on x86-64-linux and compared the outputs with diffoscope
and currently building it on aarch64-linux native

Artturin commented:
@ofborg build tests.stdenv
